JavaFXUserInterface/src/main/java/SystemsController.java

```diff
@@ -134,15 +139,54 @@ public void handle(MouseEvent mouseEvent) {
             }
         });
 
+        //Adding items to the choiceBox (drop down list)
+        sortSystem.getItems().add("Default");
+        sortSystem.getItems().add("Ascending RUL");
+        sortSystem.getItems().add("Descending RUL");
+        //Default Value
+        sortSystem.setValue("Default");
+        //Listener on the sort ChoiceBox. Depending on the sort selected, all systems panes are cleared and generated again
+        //with the appropriate sort applied.
+        sortSystem.getSelectionModel().selectedItemProperty().addListener(new ChangeListener<String>() {
+            @Override
+            public void changed(ObservableValue<? extends String> observableValue, String oldValue, String newValue) {
+                switch (newValue) {
+                    case "Ascending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Ascending RUL");
+                        }
+                        break;
+                    case "Descending RUL":
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Descending RUL");
+                        }
+                        break;
+                    default:
+                        if (thumbnailTab.isSelected()) {
+                            systemsThumbPane.getChildren().clear();
+                            generateThumbnails("Default");
+                        }
+                        break;
```

Comment:
is there not a better way to write this. looks to me like simply doing the following would be enough
`if (thumbnailTab.isSelected()) {
                              systemsThumbPane.getChildren().clear();
                              generateThumbnails(newValue);
                          }`

Reply:
in fact I would recommend doing the sorting here (see my other comments)